scripts/irods/test/rule_texts_for_tests.py

```diff
@@ -85,6 +85,42 @@
 #===== Test_Native_Rule_Engine_Plugin  =====
 
 rule_texts['irods_rule_engine_plugin-irods_rule_language']['Test_Native_Rule_Engine_Plugin'] = {}
+rule_texts['irods_rule_engine_plugin-irods_rule_language']['Test_Native_Rule_Engine_Plugin']['test_error_smsi_5043'] = '''
+validate_error_and_state_smsi(*name,*error_int) {
+    *msg = "[general_failure]"
+    *error_int = 0
+    *status = errorcode( *error_i = error(*name) )
+    if (*status != 0) {
+        *msg = "BAD error() call"
+    }
+    else {
+        *status = errorcode( *state_i = state(*name) )
+        if (*status != 0) {
+            *msg = "BAD state() call"
+        }
+        else {
+            if ((*error_i < 0 && *error_i == -*state_i) ||
```

Comment:
abs() ?

Reply:
should be opposite in sign, and nonzero
So... yes, maybe I'll replace it with something like (*error_i != 0 && abs(*error_i) == *state_i)

Update : probably better to leave as-is. abs(x) returns a DOUBLE in the iRODS rule language. Comparison of integer to integer is preferable , even if less compact.

Reply:
perfect - as is is great.